Rewrite and expand explanation of interchannel decorrelation (#98)
ietf-wg-cellar/flac-specification: flac.md

- **Left-side**. The left subblock is coded and the left and right subblock are used to code a side subframe. The side subframe is constructed in the same way as for mid-side. To decode, the right subblock is restored by subtracting the samples in the side subframe from the corresponding samples the left subframe.

- **Right-side**. The right subblock is coded and the left and right subblock are used to code a side subframe. Note that the actual coded subframe order is side-right. The side subframe is constructed in the same way as for mid-side. To decode, the left subblock is restored by adding the samples in the side subframe to the corresponding samples in the left subframe.

> The side channel needs one extra bit of bitdepth as the subtraction can produce sample values twice as large as the maximum possible in any given bitdepth. The mid channel in mid-side stereo does not need one extra bit, as it is is shifted left one bit. The left shift of the mid channel does not lead to non-lossless behaviour, because an uneven sample in the mid subframe must always be accompanied by a corresponding uneven sample in the side subframe, which means the lost least significant bit can be restored by taking it from the sample in the side subframe.

Reviewer: I guess, in Cellar we use the US syntax rather than the UK syntax: `behaviour` -> `behavior`

* The bits-per-sample bits in the `FRAME_HEADER` MUST be 0b001-0b111.
* If the sample rate is <= 48000 Hz, the filter order in `LPC subframes` (see [SUBFRAME_LPC section](#subframelpc)) MUST be less than or equal to 12, i.e. the subframe type bits in the `SUBFRAME_HEADER` (see [SUBFRAME_HEADER section](#subframeheader)) SHOULD NOT be 0b101100-0b111111.
* The Rice partition order in an `exp-golomb coded residual section` (see [RESIDUAL\_CODING\_METHOD\_PARTITIONE\_EXP\_GOLOMB section](#residualcodingmethodpartitionedexpgolomb)) MUST be less than or equal to 8.

> * The channel ordering is a standard one, i.e. no WAVEFORMATEXTENSIBLE_CHANNEL_MASK tag is necessary to define a non-standard channel ordering (see [channel assignment](#channel-assignment)).

Reviewer 2: IMO "standard one" is a bit misleading as there isn't a standard there :).

I would remove this line.

Author: What I meant is a channel ordering that differs from the ones defined under channel assignment: https://github.com/ietf-wg-cellar/flac-specification/blob/ba1e0d13f53dc1e1fed043b5ac3de39b99a4a67e/flac.md?plain=1#L381-L390 What about the following?

```suggestion
* The channel ordering is different from the ordering as defined under [channel assignment](#channel-assignment), i.e. no WAVEFORMATEXTENSIBLE_CHANNEL_MASK tag is necessary to define a channel ordering.
```

0b1010          | mid/side stereo: channel 0 is the mid(average) channel, channel 1 is the side(difference) channel
0b1011 - 0b1111 | reserved

> In case different channels than those in the table above are coded, it is recommended to add a WAVEFORMATEXTENSIBLE_CHANNEL_MASK to a VORBIS_COMMENT metadata block to describe which channels are stored. This channel mask consists of flag bits indicating which channels are present, stored in a hexadecimal representation preceded by 0x. The flags only signal which channels are present, not in which order, so in case a file has to be encoded in which channels are ordered differently, they have to be reordered. Please note that a file in which the channel order is defined through the WAVEFORMATEXTENSIBLE_CHANNEL_MASK is not streamable, i.e. non-subset, as the tag is not found in each frame header. The mask bits can be found in the following table

Reviewer 2: In that case the line in the Subset section about WAVEFORMATEXTENSIBLE_CHANNEL_MASK is also to be removed.

Your text about WAVEFORMATEXTENSIBLE_CHANNEL_MASK is really worth it and it would be a pity to lose it, could you do a dedicated PR for not losing it?

> In case different channels than those in the table above are coded, it is recommended to add a WAVEFORMATEXTENSIBLE_CHANNEL_MASK to a VORBIS_COMMENT metadata block to describe which channels are stored. This channel mask consists of flag bits indicating which channels are present, stored in a hexadecimal representation preceded by 0x. The flags only signal which channels are present, not in which order, so in case a file has to be encoded in which channels are ordered differently, they have to be reordered. Please note that a file in which the channel order is defined through the WAVEFORMATEXTENSIBLE_CHANNEL_MASK is not streamable, i.e. non-subset, as the tag is not found in each frame header. The mask bits can be found in the following table

Author: I will push a new PR after the new draft is published, under vorbis comments, together with a reference to rfc7845. I will remove the subsection bit as well, for now.

Bit number | Channel description
----------:|:-----------
0          | Front left
1          | Front right
2          | Front center

> 3          | LFE

Reviewer: As all the other channels are written out, I suggest to do the same here: `low-frequency effects`.
